bin/automatic_review.py

```diff
@@ -0,0 +1,174 @@
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser()
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--run_review', action='store_true')
+    group.add_argument('--sample_review', action='store_true')
+    return parser.parse_args()
+
+
+def query(content, parts, top_level=None, ret_default=None):
+    if top_level is None:
+        top_level = content
+    item = None
+    for p in parts:
+        item = top_level.get(p)
+        if item != None:
+            top_level = item
+        else:
+            return ret_default
+
+    return item
+
+
+def morethan(a, b):
+    if a is not None:
+        return a >= b
+    else:
+        return False
+
+def lessthan(a, b):
+
+    if a is not None:
+        return a <= b
+    else:
+        return False
+
+def inlist(a, b):
+    if a is not None:
+        return a in b
+
+def metrics(dataset, datatype):
+    PassFailDict = {}
+    return_failed_metrics = None
+
+    for metric in (cfg.get(datatype)):
+        metric_value = (query(dataset, metric.split(',')))
+        metric_name = (metric.split(',')[-1])
+        if ((cfg.get(datatype)).get(metric)['comparison']) == '>':
+            if morethan(metric_value, ((cfg.get(datatype)).get(metric)['value'])):
+                PassFailDict[metric_name] = 'pass'
+            else:
+                PassFailDict[metric_name] = 'fail'
+        elif ((cfg.get(datatype)).get(metric)['comparison']) == '<':
+            if lessthan(metric_value, ((cfg.get(datatype)).get(metric)['value'])):
+                PassFailDict[metric_name] = 'pass'
+            else:
+                PassFailDict[metric_name] = 'fail'
+        elif ((cfg.get(datatype)).get(metric)['comparison']) == 'inlist':
+            values = (cfg.get(datatype).get(metric)['value'])
+            values_list = []
+            for v in values:
+                value = (query(dataset, [v], ret_default=[v]))
+                values_list.append(''.join(value))
+            check_metric = (list(set(values_list)))
+            if inlist(metric_value, check_metric):
+                PassFailDict[metric_name] = 'pass'
+            else:
+                PassFailDict[metric_name] = 'fail'
+
+    if all(value == 'pass' for value in PassFailDict.values()):
+        return ['pass']
+    else:
+        failed_metrics = ([value for value in PassFailDict if PassFailDict[value] == 'fail'])
+        return_failed_metrics = ['fail', failed_metrics]
+    return return_failed_metrics
+
+
+class AutomaticRunReview():
+    def __init__(self, run_name, run_element_by_lane):
+        self.run_name = run_name
+        self.run_element_by_lane = run_element_by_lane
+
+    def patch_entry(self):
+        run_metrics = {}
+        for lane in self.run_element_by_lane:
+            lane_id = lane['lane_number']
+            run_metrics[lane_id] = metrics(lane, 'run')
+        if run_metrics:
+            for lane in run_metrics:
+                patch_review = {}
```

Comment:
There is no need for two independent dict for review and comment.
You should use one dict that get populated with one, in one case or two in the other case.
You would also save one patch request in the case of "fail".